pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -463,6 +463,8 @@ func (s *ServiceEntryStore) WorkloadInstanceHandler(wi *model.WorkloadInstance,
 				instancesDeleted = append(instancesDeleted, di)
 			}
 			s.serviceInstances.deleteServiceEntryInstances(seNamespacedName, key)
+		} else if event == model.EventDelete {
+			s.serviceInstances.deleteServiceEntryInstances(seNamespacedName, key)
```

Comment:
good catch, how do you find it

Reply:
Thanks @hzxuzhonghu! I found this when working on #36078

BTW, could you please help add `release-note-none` label, the [comment](https://github.com/istio/istio/pull/36674#issuecomment-1003285961) seems not work